examples/website/geospatial/app.js

```diff
@@ -0,0 +1,135 @@
+  longitude: -123.13,
+  zoom: 11,
+  maxZoom: 16,
+  pitch: 45,
+  bearing: 0
+};
+
+export default class App extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // CURRENT VIEW POINT / CAMERA POSITIO
+      viewState: INITIAL_VIEW_STATE,
+
+      // EXAMPLE STATE
+      selectedExample: INITIAL_EXAMPLE_NAME,
+      selectedLoader: GeoPackageLoader.name,
+      uploadedFile: null
+    };
+
+    this._onLoad = this._onLoad.bind(this);
+    this._onExampleChange = this._onExampleChange.bind(this);
+    this._onFileRemoved = this._onFileRemoved.bind(this);
+    this._onFileUploaded = this._onFileUploaded.bind(this);
+  }
+
+  _rotateCamera() {
+    const {viewState} = this.state;
+    this.setState({viewState});
+  }
+
+  _onViewStateChange({viewState}) {
+    this.setState({viewState});
+  }
+
+  _onExampleChange({selectedLoader, selectedExample, example}) {
+    const {data, viewState} = example;
+    this.setState({selectedLoader, selectedExample, viewState});
+    load(data).then(this._onLoad.bind(this));
+  }
+
+  _onFileRemoved(){
+   this.setState({uploadedFile: null});
+  }
+
+  _onFileUploaded(data, loader, uploadedFile){
+    this.setState({selectedLoader: loader.name, selectedExample: uploadedFile.name, uploadedFile: data});
+  }
+
+  _onLoad() {
+    const {viewState} = this.state;
+    this.setState({viewState}, this._rotateCamera);
+  }
+
+  _renderControlPanel() {
+    const {selectedExample, viewState, selectedLoader} = this.state;
+
+    return (
+      <ControlPanel
+      examples={EXAMPLES}
+      selectedExample={selectedExample}
+      selectedLoader={selectedLoader}
+      onExampleChange={this._onExampleChange}
+      >
+        <div style={{textAlign: 'center'}}>
+          long/lat: {viewState.longitude.toFixed(5)},{viewState.latitude.toFixed(5)}, zoom:{' '}
+          {viewState.zoom.toFixed(2)}
+        </div>
+        {<FileUploader onFileUploaded={this._onFileUploaded} onFileRemoved={this._onFileRemoved}/>}
+      </ControlPanel>
+    );
+  }
+
+  _renderLayer() {
+    const {selectedExample, selectedLoader, uploadedFile} = this.state;
+    debugger;
+    return [
+      new GeoJsonLayer({
+        id: `geojson-${selectedExample}(${selectedLoader})`,
+        data: uploadedFile ? uploadedFile : get(EXAMPLES, `[${selectedLoader}][${selectedExample}].data`, EXAMPLES.GeoPackage.Vancouver.data),
+        opacity: 0.8,
+        stroked: false,
+        filled: true,
+        extruded: true,
+        wireframe: true,
+        getElevation: f => Math.sqrt(f.properties.valuePerSqm) * 10,
+        getLineColor: [255, 255, 255],
+        pickable: true
+      })
+    ];
+  }
+
+  render() {
+    const {viewState} = this.state;
+
+    return (
+      <div style={{height: '100%'}}>
+        {this._renderControlPanel()}
+        <DeckGL
+          layers={this._renderLayer()}
+          viewState={viewState}
+          onViewStateChange={this._onViewStateChange.bind(this)}
```

Comment:
Nit: bind the method in constructor because you are doing this with the most of methods in this class.